File: IntegracaoCSharp/CSharp-test2/Recorrencia.cs
using System;
using PagarMe;

namespace CSharptest2
{
    public class Recorrencia
    {
        public String TRecorrencia()
        {
            PagarMeService.DefaultApiKey = "ak_test_jbIXhrHXHOaUUNKtUVrkT9HGL60SSg";

            Subscription subscription = new Subscription
            {
                PaymentMethod = PaymentMethod.CreditCard,
                CardNumber = "4901720080344448",
                CardHolderName = "Jose da Silva",
                CardExpirationDate = "1225",
                CardCvv = "123"
            };

            subscription.Plan = PagarMeService.GetDefaultService().Plans.Find(435922);
            Customer customer = new Customer
            {
                Email = "api@test.com",
                Name = "Rick",
                Documents = new[]
            {
                new Document{
                Type = DocumentType.Cpf,
                Number = "30621143049"
                }
            },
            };

            subscription.Customer = customer;

            subscription.Save();
            return Convert.ToString(subscription.Id);
        }
    }
}

File: IntegracaoRuby/recorrencia.rb
require 'pagarme'

def Recorrencia()
    
    PagarMe.api_key = "ak_test_jbIXhrHXHOaUUNKtUVrkT9HGL60SSg"

    plan = PagarMe::Plan.find_by_id("435922")

    subscription = PagarMe::Subscription.new({
        :payment_method => "credit_card",
        :card_number => "4111111111111111",
        :card_holder_name => "Jose da Silva",
        :card_expiration_month => "10",
        :card_expiration_year => "20",
        :card_cvv => "314",
        :postback_url => "http://test.com/postback",
        :customer => {
            email: "api@test.com"
        }
    })
    subscription.plan = plan

    subscription.create
end
